File: backend/models/Record.js
const mongoose = require('mongoose');

const AccessLogSchema = new mongoose.Schema({
    timestamp: Date,
    userId: String,
    action: String // 'view', 'download', 'share'
});

const RecordSchema = new mongoose.Schema({
    patientId: String,
    file: String, // File path or FHIR reference
    encrypted: Boolean,
    accessLevel: { type: String, enum: ['doctor', 'pharmacist', 'diagnostic'] },
    sharedTokens: [{ token: String, expires: Date }],
    accessLogs: [AccessLogSchema]
});

module.exports = mongoose.model('Record', RecordSchema);

File: android-app/app/src/main/java/com/healthlock/models/Record.kt
package com.healthlock.models

data class AccessLog(
    val timestamp: String,
    val userId: String,
    val action: String
)

data class Record(
    val _id: String,
    val patientId: String,
    val file: String,
    val encrypted: Boolean,
    val accessLevel: String,
    val sharedTokens: List<SharedToken>,
    val accessLogs: List<AccessLog>
)

data class SharedToken(
    val token: String,
    val expires: String
)
